File: app/src/main/java/com/example/animabuilder/CharacterCreation/Attributes/CharRace.java
package com.example.animabuilder.CharacterCreation.Attributes;

public class CharRace {
    RaceName heldRace;
    public RaceName getHeldRace(){
        return heldRace;
    }


    int raceIndex;

    public int getRaceIndex() {
        return raceIndex;
    }

    public CharRace(RaceName raceType){
        heldRace = raceType;

        switch(raceType){
            case sylvain:
                raceIndex = 1;
                break;

            case jayan:
                raceIndex = 2;
                break;

            case danjayni:
                raceIndex = 3;
                break;

            case ebudan:
                raceIndex = 4;
                break;

            case daimah:
                raceIndex = 5;
                break;

            case dukzarist:
                raceIndex = 6;
                break;

            default:
                raceIndex = 0;
        }
    }
}

File: app/src/main/java/com/example/animabuilder/CharacterCreation/Attributes/Race/CharRace.kt
package com.example.animabuilder.CharacterCreation.Attributes.Race

import com.example.animabuilder.CharacterCreation.Attributes.Race.RaceName
import java.io.Serializable

class CharRace(var heldRace: RaceName) : Serializable {
    var raceIndex = 0

    init {
        raceIndex = when (heldRace) {
            RaceName.sylvain -> 1
            RaceName.jayan -> 2
            RaceName.danjayni -> 3
            RaceName.ebudan -> 4
            RaceName.daimah -> 5
            RaceName.dukzarist -> 6
            else -> 0
        }
    }
}